kornia/geometry/transform/affwarp.py

```diff
@@ -113,6 +150,43 @@ def affine(tensor: torch.Tensor, matrix: torch.Tensor, mode: str = 'bilinear',
     return warped
 
 
+def affine3d(tensor: torch.Tensor, matrix: torch.Tensor, mode: str = 'bilinear',
+             align_corners: bool = False) -> torch.Tensor:
+    r"""Apply an affine transformation to the 3d volume.
+
+    Args:
+        tensor (torch.Tensor): The image tensor to be warped.
+        matrix (torch.Tensor): The 3x4 affine transformation matrix.
+        mode (str): 'bilinear' | 'nearest'
+        align_corners(bool): interpolation flag. Default: False. See
+        https://pytorch.org/docs/stable/nn.functional.html#torch.nn.functional.interpolate for detail
+
+    Returns:
+        torch.Tensor: The warped image.
+    """
+    # warping needs data in the shape of BCDHW
+    is_unbatched: bool = tensor.ndimension() == 4
+    if is_unbatched:
+        tensor = torch.unsqueeze(tensor, dim=0)
+
+    # we enforce broadcasting since by default grid_sample it does not
+    # give support for that
+    matrix = matrix.expand(tensor.shape[0], -1, -1)
+
+    # warp the input tensor
+    depth: int = tensor.shape[-3]
+    height: int = tensor.shape[-2]
+    width: int = tensor.shape[-1]
+    warped: torch.Tensor = warp_projective(tensor, matrix, (depth, height, width), mode,
+                                           align_corners=align_corners)
```

Comment:
I open a discussion here about naming `warp_projective` Vs `warp_affine3d` @ducha-aiki @lferraz 

Probably given the current functionality it's the second one but we could include also projection matrix adaption, which in fact I guess would be something similar to `warp_frame_depth`.

Reply:
I think, it is better to have them separated, so warp affine3d. Because affine_grid has efficient pytorch implementation and not the projectvie grid.

Reply:
great. @shijianjian feel free to rename and so.